packages/babel-parser/src/parser/expression.ts

```diff
@@ -172,7 +173,7 @@ export default abstract class ExpressionParser extends LValParser {
     expr.comments = this.comments;
     expr.errors = this.state.errors;
     if (this.optionFlags & OptionFlags.Tokens) {
-      expr.tokens = this.tokens;
+      expr.tokens = createExportedTokens(this.tokens);
```

Comment:
This change stems from the improved `ExportedTokens` typing, which clearly indicates that previously we passed the internal tokens to the parser return result.